### DISCUSSION TITLE
Limit max uploads.

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4064] [Q&A]
While working on implementing a storage function, I encountered an issue. It appears that there is no straightforward way to determine the size of a file, to set limits on user uploads. Is there a solution or a way to write a custom function to fix this?

Thanks!

**Comment by ganigeorgiev** [comment]
There is a `file` field "Max file size" option:

![file_size](https://i.imgur.com/cqtBdRq.png)

If that's not what you are looking for, please elaborate a little more on your use case and what do you mean by _"implementing a storage function"_.

**Reply by bijsven** [reply]
Thank you for your response,

But if you're dealing with multiple records, especially in a scenario where users can upload files, I want to set that when there is a UserID with 15GB on storage in a single collection, block creations.

**Reply by ganigeorgiev** [reply]
There is no builtin functionality to track total uploaded files across collections.

If you want to track storage size per some specific criteria, you can create a new collection that has a `number` field and update this field programmatically via the hooks.
The main difficulty with this is that we currently don't have a generic filesystem hooks (_this will be handled sometime in the future as part of #2387 but for now remains a low priority_) so you'll have to attach to the `OnRecordBeforeCreateRequest/OnRecordBeforeUpdateRequest` hooks and manually query the `e.UploadedFiles` slice for the newly created files + query the filesystem for the old files (you can get the old record data `e.Record.OriginalCopy()` to create/update records in the files size counter collection.
